pkg/apiserver/registry/inventory/virtualmachine/rest.go

```diff
@@ -0,0 +1,177 @@
+// Copyright 2023 Antrea Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package virtualmachine
+
+import (
+	"context"
+	"strings"
+
+	logger "github.com/go-logr/logr"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metatable "k8s.io/apimachinery/pkg/api/meta/table"
+	"k8s.io/apimachinery/pkg/apis/meta/internalversion"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/watch"
+	"k8s.io/apiserver/pkg/endpoints/request"
+	"k8s.io/apiserver/pkg/registry/rest"
+
+	runtimev1alpha1 "antrea.io/nephe/apis/runtime/v1alpha1"
+	"antrea.io/nephe/pkg/controllers/config"
+	"antrea.io/nephe/pkg/controllers/inventory"
+	"antrea.io/nephe/pkg/controllers/inventory/common"
+	"antrea.io/nephe/pkg/controllers/inventory/store"
+)
+
+// REST implements rest.Storage for VirtualMachine Inventory.
+type REST struct {
+	cloudInventory inventory.Interface
+	logger         logger.Logger
+}
+
+var (
+	_ rest.Scoper  = &REST{}
+	_ rest.Getter  = &REST{}
+	_ rest.Watcher = &REST{}
+	_ rest.Lister  = &REST{}
+)
+
+// NewREST returns a REST object that will work against API services.
+func NewREST(cloudInventory inventory.Interface, l logger.Logger) *REST {
+	return &REST{
+		cloudInventory: cloudInventory,
+		logger:         l,
+	}
+}
+
+func (r *REST) New() runtime.Object {
+	return &runtimev1alpha1.VirtualMachine{}
+}
+
+func (r *REST) NewList() runtime.Object {
+	return &runtimev1alpha1.VirtualMachine{}
+}
+
+func (r *REST) ShortNames() []string {
+	return []string{"vm"}
+}
+
+func (r *REST) Get(ctx context.Context, name string, _ *metav1.GetOptions) (runtime.Object, error) {
+	ns, ok := request.NamespaceFrom(ctx)
+	if !ok || len(ns) == 0 {
+		return nil, errors.NewBadRequest("Namespace cannot be empty.")
+	}
+
+	namespacedName := ns + "/" + name
+	vm, ok := r.cloudInventory.GetVmByKey(namespacedName)
+	if !ok {
+		return nil, errors.NewNotFound(runtimev1alpha1.Resource("virtualmachine"), name)
+	}
+	return vm, nil
+}
+
+func (r *REST) List(ctx context.Context, options *internalversion.ListOptions) (runtime.Object, error) {
+	// List only supports two types of input options
+	// 1. All namespaces
+	// 2. Labelselector with only the specific namespace, the only valid labelselectors are "cpa.name=<accountname>"
```

Comment:
update comment based on new label fields. cpa.name is not valid now.

Reply:
changing the label from cpa.name to cpa-name will be handled as part of label changes in @archanapholla's PR.